probe-builder/build-probe-binaries

```diff
@@ -0,0 +1,869 @@
+		Specify this option to skip this step and pull the images
+		from a registry instead.
+
+		The images are called <PREFIX>sysdig-probe-builder:<TAG>, where:
+		- <PREFIX> is the value of this option,
+		- <TAG> is chosen for every build. See the Dockerfiles shipped
+		  with this script.
+
+	-d DOWNLOAD_CONCURRENCY
+		The number of parallel kernel package downloads. Set to 0 to disable
+		downloads completely. Defaults to the number of CPU cores.
+
+		Note: CoreOS images are downloaded serially, regardless of the value
+		of this setting (0 is still honored).
+
+	-k KERNEL_TYPE
+		Build only for specified kernel types. The supported types are:
+			AmazonLinux
+			AmazonLinux2
+			CentOS
+			CoreOS
+			Debian
+			Fedora
+			Fedora-Atomic
+			OracleOL6
+			OracleOL7
+			OracleRHCK
+			RHEL [requires Artifactory authentication]
+			Ubuntu
+
+		If not specified, builds all the above types (RHEL kernels will be
+		skipped if no Artifactory credentials are provided).
+
+		Additionally, you can skip the downloading of distribution kernels
+		and provide your own DEBs/RPMs for the Custom* kernel types:
+
+		$0 (...) -k CustomCentOS path/*.rpm
+		$0 (...) -k CustomCoreOS path/*.bin.bz2
+		$0 (...) -k CustomDebian path/*.deb
+		$0 (...) -k CustomUbuntu path/*.deb
+
+		The CustomCentOS type should also work for other RPM-based distributions,
+		like Fedora, RHEL or AmazonLinux
+
+		For CoreOS, the downloaded images must be named
+		coreos_developer_container.VERSION.bin.bz2, where VERSION is
+		the CoreOS release (e.g. 2345.0.0).
+
+	-p PROBE_NAME
+		The resulting module name (e.g. sysdig-probe for OSS sysdig).
+
+	-r RETRIES
+		Number of retries for kernel package downloads.
+
+	-s SYSDIG_DIR
+		The directory containing an existing Sysdig checkout in the required
+		version. Specify this option to prevent this script from cloning
+		the Sysdig repository.
+
+		Note: you still need to pass -v with the right version, otherwise
+		the version number and the actual code will diverge, leading to
+		unpredictable behavior, including kernel crashes.
+
+	-t DOWNLOAD_TIMEOUT
+		Timeout for kernel package downloads, in seconds.
+
+	-v VERSION
+		The resulting module version. If not disabled (see -s), this script
+		will clone the Sysdig repository again and check out the right
+		version.
+EOF
+	exit 1
+}
+
+while getopts ":a:A:b:d:k:p:r:s:t:v:" opt
+do
+	case "$opt" in
+		a) ARTIFACTORY_SERVER=$OPTARG;;
+		A) ARTIFACTORY_KEY=$OPTARG;;
+		b) BUILDER_IMAGE_PREFIX=$OPTARG;;
+		d) DOWNLOAD_CONCURRENCY=$OPTARG;;
+		k) KERNEL_TYPE=$OPTARG;;
+		p) PROBE_NAME=$OPTARG;;
+		r) RETRIES=$OPTARG;;
+		s) SYSDIG_DIR=$OPTARG;;
+		t) DOWNLOAD_TIMEOUT=$OPTARG;;
+		v) PROBE_VERSION=$OPTARG;;
+		\?)
+			echo "Invalid option $OPTARG" >&2
+			usage
+			;;
+		:)
+			echo "Option $OPTARG requires an argument" >&2
+			usage
+			;;
+	esac
+done
+
+if [ -z "$PROBE_VERSION" ]
+then
+	echo "-v VERSION is mandatory" >&2
+	usage
+fi
+
+shift $((OPTIND - 1))
```

Comment:
Nit: Suggest you move this after the `while` loop above (before the PROBE_VERSION check).

Reply:
Not sure why but at the same time, why not :)